src/main/java/Sprites.java

```diff
@@ -16,18 +17,49 @@
      */
     public Sprites(JSONObject stage) {
         JSONArray stageChildren = stage.optJSONArray("children");
+        spriteMap = new HashMap<String, JSONObject>();
         if (stageChildren == null) {
-            stageChildren = new JSONArray();
+            return;
         }
-        spriteMap = new HashMap<String, JSONObject>();
         for (int i = 0; i < stageChildren.length(); i++) {
             JSONObject child = stageChildren.optJSONObject(i);
             if (isSprite(child)) {
-                spriteMap.put(child.getString("objName"), child);
+                addSpriteToSpriteMap(child);
             }
         }
     }
     /**
+     * Add sprite to spriteMap. Guarantee unique name.
+     * @param sprite A Sprite JSONObject
+     */
+    private void addSpriteToSpriteMap(JSONObject sprite) {
```

Comment:
Or, are we already spending too much time/energy on this case that we're not even sure that we will encounter... hrm.